src/opnsense/mvc/app/library/OPNsense/Base/UIModelGrid.php

```diff
@@ -121,9 +121,27 @@ public function fetch(
                     continue;
                 }
 
+                $flatten = function($input) {
```

Comment:
probably better to move this into the class as method, each loop (record) will reconstruct the function here. 

With a bit of magic, we should be able to save some space, how about something like:


```
function flatten($node, $path='')
{
    if (is_array($node)) {
        foreach ($node as $key => $value) {
            yield from flatten($value, ltrim($path . '.'. $key, '.'));
        }
    } else {
       yield $path => $node;
    }
}
```

In which case we can merge the payload in a one-liner `iterator_to_array(flatten($mydata))`

Reply:
@AdSchellevis Clever, neat :)

Unfortunately, the front-end can't really deal with "model.container.key" constructs without more structural changes (backwards incompatibility for non-`base_bootgrid` tables and dependencies on formatter keys etc.). The simplest solution is still to only move the key to the root, at the risk of collisions, thoughts?